src/login/logind-inhibit.c

```diff
@@ -420,11 +419,11 @@ bool manager_is_inhibited(
                 if (!(i->what & w))
                         continue;
 
-                if ((block && !IN_SET(i->mode, INHIBIT_BLOCK, INHIBIT_BLOCK_WEAK)) ||
-                    (!block && i->mode != INHIBIT_DELAY))
+                if ((flags & MANAGER_IS_INHIBITED_CHECK_DELAY) != (i->mode == INHIBIT_DELAY))
```

Comment:
this is really wrong?

it drops the INHIBIT_BLOCK/INHIBIT_BLOCK_WEAK check entirely?

and it `flags & MANAGER_IS_INHIBITED_CHECK_DELAY` returns a mask, not a bolean, hence you shouldn't compare it with a boolean. this works out of luck, because MANAGER_IS_INHIBITED_CHECK_DELAY happens to be 1, but that should not be encoded here.